### PR TITLE
Add a new docs page about testing in the package

### PR DESCRIPTION
**Pull request by shadiakiki1986**
This was split out from PR #1186. Here's a copy of my original PR description:

- added a new page in the docs about testing. I think this needs some more revision, but I just wanted to get your opinion first about having such a page in the docs. It's basically the same instructions as odow laid out in the issue. I think that it'd be a good documentation for how the testing works (I personally wasn't expecting that the optimizers ran tests from the MOI package), as well as tests being tested, and where to plug in stuff in order to add a new test. Let me know if you think it's a bad idea.

Oscar had made a first revision there as well, which is already incorporated in this PR at this stage.

This PR might need to be rebased if PRs #1186 and #1188 (or others) are merged first.

### CODE REVIEW

File: docs/src/testing.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+To run the tests:
+
+```
+julia # with 1.5
+julia> ]
+(@v1.5) pkg> dev ECOS
+(@v1.5) pkg> dev MathOptInterface
+
+Make changes in ~/.julia/dev/ECOS or ~/.julia/dev/MathOptInterface
```
Comment (odow) [inline docs/src/testing.md:77]: This bit isn't really `To run the tests:`. 

Perhaps step 1 should be `dev`ing the package, and the final step can be running the tests.

Comment (shadiakiki1986) [reply docs/src/testing.md]: done

**Comment by shadiakiki1986** [comment]
I added a final step about git commit and branch naming convention. The branch is also rebased on master.

**Comment by shadiakiki1986** [comment]
Added a link to blegat/SolverTests after learning about it in #1192

File: docs/src/testing.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# Testing
+
+All solvers use the tests in this repository as extra correctness tests for themselves.
+If we find a bug in one solver, instead of adding a test to that particular repository, we add it here so that all solvers can benefit.
+All supported solvers are tested on travis with https://github.com/blegat/SolverTests .
```
Comment (blegat) [inline docs/src/testing.md:5]: Remove space before dot

Comment (shadiakiki1986) [reply docs/src/testing.md]: done

**Pull request merged**